pkg/gui/diff_context_size.go

```diff
@@ -0,0 +1,50 @@
+package gui
+
+func getRefreshFunction(gui *Gui) func()error {
+	key := gui.currentStaticContext().GetKey()
+
+	if key == FILES_CONTEXT_KEY {
+		return gui.refreshFilesAndSubmodules
+	} else if key == MAIN_STAGING_CONTEXT_KEY {
+		return func() error {
```

Comment:
something unfortunate here is that when when we're selecting lines in a patch (i.e. press enter on a commit, then enter on one of its files and start selecting some lines with space) we're tracking the selected lines in terms of their line number in the diff as opposed to the line number in the underlying file. This means that if we change the amount of context we no longer select the same stuff. I'm not sure how easy it would be to fix this but also don't really mind releasing this feature without solving that problem because it's not that harmful.

I've also noticed that when staging a file line-by-line (i.e. press enter on a file and start selecting lines to stage) if we increase the amount of context our cursor is moved to the top of the window and range-mode is on so when you move the cursor down it selects a range. Not sure what's causing that.

One final thing is that I wonder if we can move the logic of this function into our Context modelling (admittedly confusing that this PR is about a different kind of 'context'). For example I wonder if `gui.postRefreshUpdate(gui.currentStaticContext())` works as well as this getRefreshFunction does.

Reply:
You're right about the line selection issue. I wasn't able to evaluate this properly, though, because the other issue you found (the cursor jumping to the top when changing context size in patch or staging) cover this up for me---or is there another way to scroll down a diff from, say, the list of changed files? Either way, I'm still looking at this cursor jumping issue.

I've tried `gui.postRefreshUpdate(gui.currentStaticContext())` and it does work, except in patch or staging. Afaict, `postRefreshUpdate` calls `HandleRender` which only seems to be available on `ListContext` objects. Does that sound right? But I can definitely use that to make `getRefreshFunction` much easier.

Reply:
Update: it seems that [these lines](https://github.com/jesseduffield/lazygit/blob/master/pkg/gui/lbl/state.go#L39) are responsible for switching to range-mode when using my shortcuts in patch or staging.

Reply:
Another update: the cursor jumping to the first line is actually my doing. If I get the selected line index from `gui.State.Panels.LineByLine` instead of `gui.Views.Secondary` it works as expected. Working on a fix (plus tests) now.